pkg/framework/core/allure_manager/ctx/test_ctx_test.go

```diff
@@ -35,3 +35,147 @@ func TestTestCtx_AddAttachment(t *testing.T) {
+
+			testCtx := NewTestCtx(result)
+
+			retrievedResult := testCtx.GetTestResult()
+			require.NotNil(t, retrievedResult)
+			require.Equal(t, tc.status, retrievedResult.Status)
+		})
+	}
+}
+
+func TestTestCtx_GetTestResult_NilResult(t *testing.T) {
+	testCtx := testCtx{name: constants.TestContextName, result: nil}
+
+	retrievedResult := testCtx.GetTestResult()
+	require.Nil(t, retrievedResult)
+}
+
+func TestTestCtx_GetName_TestContext(t *testing.T) {
+	result := allure.NewResult("TestName", "TestFullName")
+	testCtx := NewTestCtx(result)
+
+	name := testCtx.GetName()
+	require.Equal(t, constants.TestContextName, name)
+}
+
+// Additional tests for parametrized and nested test scenarios
+func TestTestCtx_GetTestResult_ParametrizedTest(t *testing.T) {
+	// Simulate parametrized test result
+	result := allure.NewResult("TestParametrized", "suite.TestParametrized")
+	result.Status = allure.Passed
+
+	testCtx := NewTestCtx(result)
+
+	retrievedResult := testCtx.GetTestResult()
+	require.NotNil(t, retrievedResult)
+	require.Equal(t, allure.Passed, retrievedResult.Status)
+	require.Equal(t, "TestParametrized", retrievedResult.Name)
+}
+
+func TestTestCtx_GetTestResult_NestedTest(t *testing.T) {
+	// Simulate nested test result
+	result := allure.NewResult("TestNested", "suite.TestNested")
+	result.Status = allure.Passed
+
+	testCtx := NewTestCtx(result)
+
+	retrievedResult := testCtx.GetTestResult()
+	require.NotNil(t, retrievedResult)
+	require.Equal(t, allure.Passed, retrievedResult.Status)
+}
+
+func TestTestCtx_GetTestResult_WithStatusDetails(t *testing.T) {
+	result := allure.NewResult("TestWithDetails", "suite.TestWithDetails")
+	result.Status = allure.Failed
+	result.SetStatusMessage("Assertion failed: expected 1, got 2")
+	result.SetStatusTrace("trace stack here")
+
+	testCtx := NewTestCtx(result)
+
+	retrievedResult := testCtx.GetTestResult()
+	require.NotNil(t, retrievedResult)
+	require.Equal(t, allure.Failed, retrievedResult.Status)
+	require.Equal(t, "Assertion failed: expected 1, got 2", retrievedResult.GetStatusMessage())
+	require.Equal(t, "trace stack here", retrievedResult.GetStatusTrace())
+}
+
+func TestTestCtx_GetTestResult_BrokenTest(t *testing.T) {
+	result := allure.NewResult("BrokenTest", "suite.BrokenTest")
+	result.Status = allure.Broken
+	result.SetStatusMessage("Panic occurred")
+
+	testCtx := NewTestCtx(result)
+
+	retrievedResult := testCtx.GetTestResult()
+	require.NotNil(t, retrievedResult)
+	require.Equal(t, allure.Broken, retrievedResult.Status)
+	require.Contains(t, retrievedResult.GetStatusMessage(), "Panic occurred")
+}
+
+func TestTestCtx_GetTestResult_SkippedTest(t *testing.T) {
+	result := allure.NewResult("SkippedTest", "suite.SkippedTest")
+	result.Status = allure.Skipped
+	result.SetStatusMessage("Test skipped")
+
+	testCtx := NewTestCtx(result)
+
+	retrievedResult := testCtx.GetTestResult()
+	require.NotNil(t, retrievedResult)
+	require.Equal(t, allure.Skipped, retrievedResult.Status)
+}
```

Comment:
These tests look identical and seem to test the same functionality

Reply:
Done